src/agent.c

```diff
@@ -580,6 +616,34 @@ void agent_run(juice_agent_t *agent) {
 					snprintf(entry->turn->credentials.username, STUN_MAX_USERNAME_LEN, "%s",
 					         turn_server->username);
 					entry->turn->password = turn_server->password;
+
+					transport_socket_config_t socket_config;
+					memset(&socket_config, 0, sizeof(socket_config));
+
+					entry->transport = turn_server->transport;
+				    if (entry->transport) {
+						socket_config.bind_address = agent->config.bind_address;
+						socket_config.port_begin = agent->config.local_port_range_begin;
+						socket_config.port_end = agent->config.local_port_range_end;
+						switch (turn_server->transport) {
+							case JUICE_TRANSPORT_UDP:
```

Comment:
What does UDP transport mean for the user? If I get it correctly, it opens a dedicated UDP socket to reach the server instead of using the one of the agent. What is the point of this mechanism?

Reply:
In my mind, the ideal is to replace the udp .c completely with transport .c， UDP, TCP and TLS are supported.

Reply:
I see. While this is smart for factoring the code, it will cause issues, for instance reflexive candidates from TURN UDP servers will be incorrect, as they'll reflect the external address for the transport UDP socket instead of the agent one. I think the only way to keep them working is limiting the transport to TCP and use the agent socket for UDP, sadly.